plugin/documents.py

```diff
@@ -494,7 +506,15 @@ def _register_async(self) -> None:
         file_name = self.view.file_name()
         if file_name:
             self._file_name = file_name
-            self.manager.register_listener_async(self)
+            buf = self.view.buffer()
+            if buf is not None:
+                text_change_listener = TextChangeListener.ids_to_listeners.get(buf.buffer_id)
+                if text_change_listener is not None:
```

Comment:
Why such verbose if-check (also for the buffer above)? Since we are dealing with class instances, an `if foo` should be enough (unless it implements some weird `__bool__ ` I guess but can we assume that it's not?).